examples/editor-assignment/src/locations/ConfigScreen.tsx

```diff
@@ -0,0 +1,174 @@
+  const [supportedFieldCTs, setSupportedFieldCTs] = useState<ContentTypeProps[]>([]);
+  const [selectedFieldCTs, setSelectedFieldCTs] = useState<string[]>([]);
+  const sdk = useSDK<AppExtensionSDK>();
+  const cma = useCMA();
+
+  const onConfigure = useCallback(async () => {
+    // This method will be called when a user clicks on "Install"
+    // or "Save" in the configuration screen.
+    // for more details see https://www.contentful.com/developers/docs/extensibility/ui-extensions/sdk-reference/#register-an-app-configuration-hook
+
+    // Get current the state of EditorInterface and other entities
+    // related to this app installation
+    const currentState = await sdk.app.getCurrentState();
+
+    return {
+      // Parameters to be persisted as the app configuration.
+      parameters,
+      // In case you don't want to submit any update to app
+      // locations, you can just pass the currentState as is
+      targetState: {
+        EditorInterface: {
+          ...currentState?.EditorInterface,
+
+          // apply to all selected content types' sidebar & fields
+          ...merge(
+            buildSidebarTargetState(selectedSidebarCTs),
+            buildFieldTargetState(selectedFieldCTs, supportedFieldCTs)
+          ),
+        },
+      },
+    };
+  }, [parameters, sdk, selectedSidebarCTs, selectedFieldCTs, supportedFieldCTs]);
+
+  const onSidebarContentTypeClick = useCallback(
+    onCTSelect(selectedSidebarCTs, setSelectedSidebarCTs),
+    [selectedSidebarCTs, setSelectedSidebarCTs]
+  );
+  const onFieldContentTypeClick = useCallback(onCTSelect(selectedFieldCTs, setSelectedFieldCTs), [
+    selectedFieldCTs,
+    setSelectedFieldCTs,
+  ]);
+
+  useEffect(() => {
+    // `onConfigure` allows to configure a callback to be
+    // invoked when a user attempts to install the app or update
+    // its configuration.
+    sdk.app.onConfigure(() => onConfigure());
+  }, [sdk, onConfigure]);
+
+  useEffect(() => {
+    (async () => {
+      // Get current parameters of the app.
+      // If the app is not installed yet, `parameters` will be `null`.
+      const currentParameters: AppInstallationParameters | null = await sdk.app.getParameters();
+
+      if (currentParameters) {
+        setParameters(currentParameters);
+      }
+
+      // Once preparation has finished, call `setReady` to hide
+      // the loading screen and present the app to a user.
+      sdk.app.setReady();
+    })();
+  }, [sdk]);
+
+  useEffect(() => {
+    (async () => {
+      const cts = await cma.contentType.getMany({
+        spaceId: sdk.ids.space,
+        environmentId: sdk.ids.environment,
+      });
+      setContentTypes(cts.items || []);
+      setSupportedFieldCTs(filterShortTextFieldCTs(cts.items));
+    })();
+  }, [cma, sdk]);
+
+  // Get all editor interfaces to set initial list of content types
+  // with app already assigned to it
+  useEffect(setInitialSidebarContentTypes(cma, sdk, setSelectedSidebarCTs), [
```

Comment:
I would unwrap this `setInitialSidebarContentTypes` function, because it includes something that usually belongs to `useEffect`. So this looks not 'standard' to me.